.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+name: ci
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+jobs:
+  main:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        type: [alpine, buster-slim]
+        node: [12, 14, 16]
+    env:
+      AZURE_CONTAINER_REGISTRY_URL: hmctspublic.azurecr.io
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v2
+      - name: Hadolint Dockerfile
+        # TODO https://github.com/hadolint/hadolint-action/pull/45
+        uses: mblottiere/hadolint-action@390bcfc1bcbdbe0eab20a313ad47a2d297a69441
+        with:
+          dockerfile: ./${{ matrix.type }}/Dockerfile
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Login to Azure Container Repository
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.AZURE_CONTAINER_REGISTRY_URL }}
+          username: ${{ secrets.AZURE_CLIENT_ID }}
+          password: ${{ secrets.AZURE_CLIENT_SECRET }}
+      - name: Build - Node ${{ matrix.node }} ${{ matrix.type }}
```

Comment:
don't forget to duplicate this with platforms and push readded after the test section if this works